### PR TITLE
Commit changes that are *hot* in the current regress directories.

### PR DESCRIPTION
**Pull request by KineticTheory**
+ `ccscs-regress.msub`:
  - On ccscs7, set `OMP_NUM_THREADS=8`
  - Force loading superlu-dist/4.3.to avoid loading 5.X.
  - Add `$featurebranch` to the build name.
+ Prevent `checkpr.sh` from running on ccscs7 unless run by the regress account. This should reduce the load on ccscs7.
+ Move *fulldiagnostics* CI build from moonlight to snow.
+ Begin running a *Release* CI build on snow.
+ Improve how `selectscratchdir` works.
+ Reactivate the repository synchronization used by Redmine.
* [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [ ] Travis CI checks pass
  * [ ] Code coverage does not decrease
  * [ ] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] [Toss2 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

File: regression/scripts/common.sh [hunk 1]
```diff
@@ -238,17 +239,17 @@ function selectscratchdir
     # if this location is good (must be able to write to this location), return
     # the path.
     mkdir -p $item/$USER &> /dev/null
-    touch $item/$USER/selectscratchdir &> /dev/null
-    if [[ -f $item/$USER/selectscratchdir ]]; then
-      rm $item/$USER/selectscratchdir
+    touch $item/$USER/selectscratchdir-${platform} &> /dev/null
+    if [[ -f $item/$USER/selectscratchdir-${platform} ]]; then
+      rm $item/$USER/selectscratchdir-${platform} &> /dev/null
```
Comment (RyanWollaeger) [inline regression/scripts/common.sh]: Would `if [[ -w $item/$USER ]]; then echo "$item"; return; fi` work here as well?

File: regression/checkpr.sh [hunk 2]
```diff
@@ -96,12 +96,15 @@ case $project in
...
 esac
 
-
-if ! [[ $LOGNAME == "kellyt" ]]; then
-  echo ""; echo "FATAL ERROR: Please use ccscs6 for manual use of checkpr.sh."
-  exit 1
-fi
-
+# Restrict the use of ccscs7.
+case $target in
```
Comment (RyanWollaeger) [inline regression/checkpr.sh]: Are more cases anticipated here (ccscs[2-3])?

File: regression/checkpr.sh [hunk 3]
```diff
@@ -230,10 +233,14 @@ case $target in
     ;;
 
   # Moonlight: Fulldiagnostics (Debug)
-  ml-fey*) startCI ${project} Debug fulldiagnostics $pr ;;
+  ml-fey*) startCI ${project} Debug $pr ;;
 
   # Snow: Debug
-  sn-fe*) startCI ${project} Debug na $pr ;;
+  sn-fe*)
+    startCI ${project} Debug na $pr
```
Comment (RyanWollaeger) [inline regression/checkpr.sh]: Is this debug build test needed with `fulldiagnostics` as well?

File: regression/scripts/common.sh (hunk 1 above)
Comment (KineticTheory) [inline regression/scripts/common.sh]: I've never used the `-w` option.  Let me try it before we merge this PR.

File: regression/checkpr.sh (hunk 2 above)
Comment (KineticTheory) [inline regression/checkpr.sh]: Possibly - there is a ticket to move some of the CI work load to another machine (maybe ccscs6).

File: regression/checkpr.sh (hunk 3 above)
Comment (KineticTheory) [inline regression/checkpr.sh]: Debug and fulldiagnostics have different logic paths in the code.  We might be able to remove a Debug or Release build from Moonlight instead.  For now, I will leave this as is.

**Comment by RyanWollaeger** [comment]
There appear to be no issues:
https://rtt.lanl.gov/cdash/index.php?project=Draco

Should the "needs-work" label be interpreted as work in progress?
Otherwise, this appears to be ready to approve and merge.

**Comment by KineticTheory** [comment]
@RyanWollaeger Labels are fixed. This is ready to be merged.

**Pull request merged**